tree-sitter/src/TreeSitter/Node.hs

```diff
@@ -18,20 +20,25 @@ data Node = Node
   { nodeTSNode     :: !TSNode
   , nodeType       :: !CString
   , nodeSymbol     :: !TSSymbol
-  , nodeStartPoint :: !TSPoint
   , nodeEndPoint   :: !TSPoint
-  , nodeStartByte  :: !Word32
   , nodeEndByte    :: !Word32
   , nodeChildCount :: !Word32
+  , nodeFieldName  :: !CString
   , nodeIsNamed    :: !CBool
   , nodeIsExtra    :: !CBool
   }
   deriving (Show, Eq, Generic)
 
+nodeStartPoint :: Node -> TSPoint
+nodeStartPoint node = let TSNode _ p _ _ _ = nodeTSNode node in p
+
+nodeStartByte :: Node -> Word32
+nodeStartByte node = let TSNode b _ _ _ _ = nodeTSNode node in b
+
```

Comment:
On @maxbrunsfeld’s recommendation, we’re now getting these out of the `TSNode` instead of copying them twice.